Simplified/NYPLAnnouncementManager.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+private let announcementKey: String = "NYPLPresentedAnnouncementKey"
+private let filename: String = "NYPLPresentedAnnouncementsList"
+
+class NYPLAnnouncementManager {
+  private static var presentedAnnouncement: Set<String> = Set<String>()
+    
+  class func presentAnnouncements(_ announcements: [Announcement]) {
+    for announcement in announcements {
+      if shouldPresentAnnouncement(id: announcement.id) {
+        DispatchQueue.main.async {
+          let vc = NYPLAnnouncementViewController(announcement: announcement)
+          NYPLRootTabBarController.shared()?.safelyPresentViewController(vc, animated: true, completion: nil)
+        }
+      }
+    }
+  }
+  
+  // MARK: - Read
+    
+  class func shouldPresentAnnouncement(id: String) -> Bool {
+    guard let filePathURL = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first?.appendingPathComponent(filename),
+      let filePathData = try? Data(contentsOf: filePathURL),
+      let unarchived = try? NSKeyedUnarchiver.unarchiveTopLevelObjectWithData(filePathData),
+      let presented = unarchived as? Set<String> else {
+        return true
```

Comment:
shouldn't this be false? or is it that if the file is not found it means it's a new announcement? but even in that case if the file is found and decoding fails, seems like we should return `false`... I could be wrong though. Some comments would be really useful here to explain the intended logic.

Reply:
Since we are making `NYPLAnnouncementBusinessLogic` a singleton, I have moved this part of code into the init.
So we only read the file once the singleton being created, and we will use the property `presentedAnnouncements` to keep track of the announcements for the life cycle.
So if the file is not found, cannot be decoded or corrupted, we will overwrite the file with the new presented announcements set.